src/forms/progressive-profiling-popup/index.jsx

```diff
@@ -72,14 +74,15 @@ const ProgressiveProfilingForm = () => {
       // set formData state for auto populated country field to pass into payload
       setFormData({ country: userCountry?.code });
     }
-  }, [authContextCountryCode, autoFilledCountry, countryList, formatMessage]);
+  }, [authContextCountryCode, autoFilledCountry, countryCookieValue, countryList, formatMessage]);
 
   useEffect(() => {
     if (authenticatedUser === null) {
       dispatch(setCurrentOpenedForm(LOGIN_FORM));
     }
     if (authenticatedUser?.userId) {
       configureAuth(AxiosJwtAuthService, { config: getConfig() });
+      dispatch(getUserProfile(authenticatedUser?.username));
```

Comment:
why do we need to fetch user profile? Don't we already know from the register page that the country field is available or not?
* If country field was available in cookie or MFE context then the account must have been created using the country field
* We can use that country field here on progressive profiling

Reply:
Initially, I was thinking about the approach you mentioned but to make sure that the user country is set in the user profile we are fetching the user account info.